const e=JSON.parse('{"key":"v-d0a7fa8e","path":"/en/essay/how-to-catch-error.html","title":"Error handling summary","lang":"en-US","frontmatter":{"title":"Error handling summary","description":"Error handling summary This is a post that I wrote on May 23, 2019, while working on a project and thinking about how to handle errors. Purpose of the article Error handling is to prevent software from causing major accidents due to abnormal behavior. In service operation, it can detect whether users are using the service abnormally.","head":[["link",{"rel":"alternate","hreflang":"ko-kr","href":"https://the-next-web-research-lab.github.io/essay/how-to-catch-error.html"}],["meta",{"property":"og:url","content":"https://the-next-web-research-lab.github.io/en/essay/how-to-catch-error.html"}],["meta",{"property":"og:site_name","content":"TheNextWebResearchLab"}],["meta",{"property":"og:title","content":"Error handling summary"}],["meta",{"property":"og:description","content":"Error handling summary This is a post that I wrote on May 23, 2019, while working on a project and thinking about how to handle errors. Purpose of the article Error handling is to prevent software from causing major accidents due to abnormal behavior. In service operation, it can detect whether users are using the service abnormally."}],["meta",{"property":"og:type","content":"article"}],["meta",{"property":"og:image","content":"https://the-next-web-research-lab.github.io/images/og_image.png"}],["meta",{"property":"og:locale","content":"en-US"}],["meta",{"property":"og:locale:alternate","content":"ko-KR"}],["meta",{"property":"og:updated_time","content":"2024-01-18T07:15:00.000Z"}],["meta",{"name":"twitter:card","content":"summary_large_image"}],["meta",{"name":"twitter:image:alt","content":"Error handling summary"}],["meta",{"property":"article:modified_time","content":"2024-01-18T07:15:00.000Z"}],["script",{"type":"application/ld+json"},"{\\"@context\\":\\"https://schema.org\\",\\"@type\\":\\"Article\\",\\"headline\\":\\"Error handling summary\\",\\"image\\":[\\"https://the-next-web-research-lab.github.io/images/og_image.png\\"],\\"dateModified\\":\\"2024-01-18T07:15:00.000Z\\",\\"author\\":[]}"]]},"headers":[{"level":2,"title":"Purpose of the article","slug":"purpose-of-the-article","link":"#purpose-of-the-article","children":[]},{"level":2,"title":"Table of contents","slug":"table-of-contents","link":"#table-of-contents","children":[]},{"level":2,"title":"How to deliver failure","slug":"how-to-deliver-failure","link":"#how-to-deliver-failure","children":[{"level":3,"title":"Delivering failure with return value","slug":"delivering-failure-with-return-value","link":"#delivering-failure-with-return-value","children":[]},{"level":3,"title":"try-catch error handling","slug":"try-catch-error-handling","link":"#try-catch-error-handling","children":[]},{"level":3,"title":"throw explicit declaration","slug":"throw-explicit-declaration","link":"#throw-explicit-declaration","children":[]}]},{"level":2,"title":"How to handle errors","slug":"how-to-handle-errors","link":"#how-to-handle-errors","children":[{"level":3,"title":"Container","slug":"container","link":"#container","children":[]},{"level":3,"title":"Return the same type","slug":"return-the-same-type","link":"#return-the-same-type","children":[]},{"level":3,"title":"Error handling with async function","slug":"error-handling-with-async-function","link":"#error-handling-with-async-function","children":[]}]},{"level":2,"title":"Comprehensive conclusion","slug":"comprehensive-conclusion","link":"#comprehensive-conclusion","children":[]},{"level":2,"title":"End","slug":"end","link":"#end","children":[{"level":3,"title":"References","slug":"references","link":"#references","children":[]}]}],"git":{"createdTime":1705562100000,"updatedTime":1705562100000,"contributors":[{"name":"ChoDragon9","email":"04whdydrn30@gmail.com","commits":1}]},"readingTime":{"minutes":4.06,"words":1219},"filePathRelative":"en/essay/how-to-catch-error.md","localizedDate":"January 18, 2024","excerpt":"<h1> Error handling summary</h1>\\n<blockquote>\\n<p>This is a post that I wrote on May 23, 2019, while working on a project and thinking about how to handle errors.</p>\\n</blockquote>\\n<h2> Purpose of the article</h2>\\n<p><strong>Error handling</strong> is to <strong>prevent</strong> software from causing major accidents due to abnormal behavior.\\nIn service operation, it can detect whether users are using the service abnormally.</p>","autoDesc":true}');export{e as data};
